### PR TITLE
Cors, mocking, dns config reference

### PR DESCRIPTION
**Pull request by Guaris**
### Preview Links

https://deploy-preview-173--kongdeveloper.netlify.app/plugins/cors/
https://deploy-preview-173--kongdeveloper.netlify.app/plugins/mocking/
https://deploy-preview-173--kongdeveloper.netlify.app/gateway/networking/dns-config-reference/

### Checklist 

- [x] Every page is page one
- [x] Tested how-to docs. If not, note why here. 
- [x] All pages contain metadata
- [x] Updated [sources.yaml](https://github.com/Kong/developer.konghq.com/blob/main/tools/track-docs-changes/config/sources.yml). For more info review [track docs changes](https://github.com/Kong/developer.konghq.com/tree/main/tools/track-docs-changes)
- [x] Any new docs link to existing docs.

### CODE REVIEW

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]
Left some suggestions, questions, and a few observations as I was reading the content. Let me know if you have any questions!

File: app/_includes/sections/cors-and-kong-gateway.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+
+## Understanding CORS
```
[the remark this answers is not in the document]
Comment (Guaris) [reply app/_includes/sections/cors-and-kong-gateway.md:2]: The example is under this with the headers being rejected.  CORS ensures that KGW doesn't deny anyone access to a resource in a proxied upstream.

File: app/_kong_plugins/mocking/index.md [hunk 2]
```diff
@@ -22,6 +23,73 @@ works_on:
...
+
+## Mocked responses
+
+The Mocking plugin can mock the following responses: 
+
+* **`200`**
+* **`201`**
+* **`204`**
+
+## Behavioral Headers
```
[the remark this answers is not in the document]
Comment (Guaris) [reply app/_kong_plugins/mocking/index.md:47]: I'm not a fan of it in this case. Maybe on a landing page.

**Review by cloudjumpercat — APPROVED** [review]

File: app/_kong_plugins/mocking/examples/basic-configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+title: Pass an OpenAPI spec to the plugin through decK
+description: |
+  Configures the mocking plugin and passes an API spec.
```
Comment (fabianrbz) [inline app/_kong_plugins/mocking/examples/basic-configuration.yaml:3]:
```suggestion
  Configures the Mocking plugin and passes an API spec.
```

File: app/_kong_plugins/cors/examples/basic-configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+title: Configure the plugin on a Service
```
Comment (fabianrbz) [inline app/_kong_plugins/cors/examples/basic-configuration.yaml:1]: The targets are set by the platform based on the plugin's support, so there's no need to specify it here or [here](https://github.com/Kong/developer.konghq.com/pull/173/files#diff-5e3f90dd52aa9a4d0e8c32238de99cd3a50f4ab2fb414e9a3062d14cac230931R10).

I guess the goal of this is to allow requests from a specific domain (http://mockbin.com) and only two methods.
We could drop the headers and go with the default ones to make it simpler and rename the file, title and description to something like `Allow requests from a specific domain`

File: app/_kong_plugins/cors/examples/basic-configuration.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      - Accept-Version
+      - Content-Length
+      - Content-MD5
+      - Content-Type
+      - Date
+      - X-Auth-Token
+      exposed_headers:
+      - X-Auth-Token
+      credentials: true
+      max_age: 3600
```
Comment (fabianrbz) [inline app/_kong_plugins/cors/examples/basic-configuration.yaml:28]:
```suggestion
config:
  origins:
    - http://mockbin.com
  methods:
    - GET
    - POST
  headers:
    - Accept
    - Accept-Version
    - Content-Length
    - Content-MD5
    - Content-Type
    - Date
    - X-Auth-Token
  exposed_headers:
    - X-Auth-Token
  credentials: true
  max_age: 3600
```

you only need the plugin's config here

File: app/_kong_plugins/mocking/examples/basic-configuration.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                content:
+                  application/json:
+                    schema:
+                      $ref: '#/components/schemas/Pet'
+                  application/xml:
+                    schema:
+                      $ref: '#/components/schemas/Pet'
+                  application/x-www-form-urlencoded:
+                    schema:
+                      $ref: '#/components/schemas/Pet'
```
Comment (fabianrbz) [inline app/_kong_plugins/mocking/examples/basic-configuration.yaml:125]:
```suggestion
config:
  random_delay: true
  max_delay_time: 1
  min_delay_time: 0.001
  random_examples: true
  api_specification: |
    openapi: 3.0.2
    servers:
      - url: /v3
    info:
      description: |-
        This is a sample Pet Store Server based on the OpenAPI 3.0 specification.  You can find out more about
        Swagger at [http://swagger.io](http://swagger.io). In the third iteration of the pet store, we've switched to the design first approach!
        You can now help us improve the API whether it's by making changes to the definition itself or to the code.
        That way, with time, we can improve the API in general, and expose some of the new features in OAS3.
        Some useful links:
        - [The Pet Store repository](https://github.com/swagger-api/swagger-petstore)
        - [The source API definition for the Pet Store](https://github.com/swagger-api/swagger-petstore/blob/master/src/main/resources/openapi.yaml)
      version: 1.0.20-SNAPSHOT
      title: Swagger Petstore - OpenAPI 3.0
      termsOfService: 'http://swagger.io/terms/'
      contact:
        email: apiteam@swagger.io
      license:
        name: Apache 2.0
        url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
    tags:
      - name: pet
        description: Everything about your Pets
        externalDocs:
          description: Find out more
          url: 'http://swagger.io'
      - name: store
        description: Access to Petstore orders
        externalDocs:
          description: Find out more about our store
          url: 'http://swagger.io'
      - name: user
        description: Operations about user
    paths:
      /pet:
        post:
          tags:
            - pet
          summary: Add a new pet to the store
          description: Add a new pet to the store
          operationId: addPet
          responses:
            '200':
              description: Successful operation
              content:
                application/xml:
                  schema:
                    $ref: '#/components/schemas/Pet'
                application/json:
                  schema:
                    $ref: '#/components/schemas/Pet'
            '405':
              description: Invalid input
          security:
            - petstore_auth:
                - 'write:pets'
                - 'read:pets'
          requestBody:
            description: Create a new pet in the store
            required: true
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Pet'
              application/xml:
                schema:
                  $ref: '#/components/schemas/Pet'
              application/x-www-form-urlencoded:
                schema:
                  $ref: '#/components/schemas/Pet'
        put:
          tags:
            - pet
          summary: Update an existing pet
          description: Update an existing pet by Id
          operationId: updatePet
          responses:
            '200':
              description: Successful operation
              content:
                application/xml:
                  schema:
                    $ref: '#/components/schemas/Pet'
                application/json:
                  schema:
                    $ref: '#/components/schemas/Pet'
            '400':
              description: Invalid ID supplied
            '404':
              description: Pet not found
            '405':
              description: Validation exception
          security:
            - petstore_auth:
                - 'write:pets'
                - 'read:pets'
          requestBody:
            description: Update an existent pet in the store
            required: true
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Pet'
              application/xml:
                schema:
                  $ref: '#/components/schemas/Pet'
              application/x-www-form-urlencoded:
                schema:
                  $ref: '#/components/schemas/Pet'
```

File: app/_kong_plugins/mocking/examples/basic-configuration.yaml [hunk 7]
```diff
@@ -0,0 +1,125 @@
+title: Pass an OpenAPI spec to the plugin through decK
+description: |
```
Comment (fabianrbz) [inline app/_kong_plugins/mocking/examples/basic-configuration.yaml:2]: The goal of this is actually to mock endpoints using an OpenAPI spec.
Can you update the file name, title and description accordingly? (something along those lines)

deck doesn't need to be part of the title, the platform generates

File: app/_kong_plugins/mocking/examples/mock-endpoints-using-an-api-spec.yaml [hunk 8]
```diff
@@ -0,0 +1,116 @@
+title: Mock endpoints using an API spec
+description: |
+  Configures the Mocking plugin and passes a the [PetStore](https://petstore3.swagger.io/) API specification.
```
Comment (Guaris) [inline app/_kong_plugins/mocking/examples/mock-endpoints-using-an-api-spec.yaml:3]:
```suggestion
  Configures the Mocking plugin and passes the [PetStore](https://petstore3.swagger.io/) API specification.
```

**Review by fabianrbz — APPROVED** [review]

**Pull request merged**